bin/analysercli

```diff
@@ -0,0 +1,274 @@
+#! /usr/bin/env python
+#
+# This file is part of IVRE.
+# Copyright 2011 - 2015 Pierre LALET <pierre.lalet@cea.fr>
+#
+# IVRE is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# IVRE is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with IVRE. If not, see <http://www.gnu.org/licenses/>.
+
+from ivre import webutils
+from ivre.db import db
+from ivre.cliutils import colorize_log, StackCli
+from ivre.analyser import Context
+
+
+class AnalyserCli(StackCli):
+    """
+    Command line utility wrapper to access analyser API
+    """
+    def __init__(self):
+        StackCli.__init__(self)
+        self._dfilter = {}
+
+    def preloop(self):
+        context = Context()
+        colorize_log('info', 'Initialization')
+        self._build_filter(raw_input('Filter > '))
+        colorize_log('info', 'Reading database..')
+        context.dfilter = self._dfilter
+        context.init_hosts()
+        self.push(context)
+
+    def postloop(self):
+        colorize_log('info', 'Leaving..')
+
+    @staticmethod
+    def _flt_param(param):
+        """
+        Return a tuple (neg, param), where `neg` is True if the param was
+        negative (i.e. starts with '!') and `param` is the argument stripped
+        from the potential heading '!'
+        """
+        return (True, param[1:]) if param.startswith('!') else (False, param)
+
+    def _build_filter(self, cmd):
+        """Build a db.nmap.flt from the string `cmd`"""
+        query = []
+        cmd = cmd.split()
+        while cmd:
+            param = cmd.pop(0)
+            neg, param = self._flt_param(param)
+            value = cmd.pop(0)
+            if param == 'limit':
+                self._dfilter['limit'] = int(value)
+            else:
+                query.append((neg, param, value))
+        (self._dfilter['filter'], _, _, _,
+         self._dfilter['skip'], _) = webutils.flt_from_query(
+             query, base_flt=db.nmap.flt_empty)
+
+    def do_count(self, _):
+        """Print host count to stdout"""
+        log = 'Number of hosts:\n'
```

Comment:
Here again to avoid `+` on strings:
```Python
colorize_log('info', "Number of hosts:\n%s" % ''.join(
    '\t%s:\t%s\n' % (key, value)
    for key, value in self.get_context().count().iteritems()
))
```
or (better?):
```Python
colorize_log('info', "Number of hosts:\n\t%s\n" % '\n\t'.join(
    '%s:\t%s' % (key, value)
    for key, value in self.get_context().count().iteritems()
))
```